src/components/games/gamequeue.hpp

```diff
@@ -0,0 +1,44 @@
+#ifndef GAME_QUEUE_CLASS_HPP
+#define GAME_QUEUE_CLASS_HPP
+
+#include "../component.hpp"
+
+namespace qb
+{
+class Queue {
+    public:
+    Queue(const std::string name, api::Message message, const std::string game, int max_size) 
+        : name_(name), message_(message), game_(game), max_size_(max_size) {}
+    
+
+    std::string to_str();
+    std::string name_;
+    api::Message message_;
+    std::string game_;
```

Comment:
worth noting that things should be `const` here when it reflects what the data should be doing/how it should be behaving

here, in my opinion, `name_` and `message_` should both be marked `const`, especially `message_`

also, you should have `std::string to_str() const;` this indicates that `to_str()` shouldn't edit any member variables (which it really shouldn't, so...) and potentially finds bugs (like if adding `const` causes compile failure, there's something wonky, or at least something to think about)